hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/server/api/protocolrecords/CheckForDecommissioningNodesRequest.java

```diff
@@ -32,4 +34,31 @@ public static CheckForDecommissioningNodesRequest newInstance() {
         .newRecord(CheckForDecommissioningNodesRequest.class);
     return request;
   }
+
+  @Private
+  @Unstable
+  public static CheckForDecommissioningNodesRequest newInstance(String subClusterId) {
+    CheckForDecommissioningNodesRequest request = Records
+        .newRecord(CheckForDecommissioningNodesRequest.class);
+    request.setSubClusterId(subClusterId);
+    return request;
+  }
+
+  /**
+   * Get the subClusterId.
+   *
+   * @return subClusterId.
+   */
+  @InterfaceAudience.Public
```

Comment:
Let's be consistent, either just Public or the whole thing.

Reply:
Thank you for your help to review the code! I will modify the code.